### PR TITLE
Use cancel token to stop server

### PR DESCRIPTION
**Pull request by will-lynas**

### CODE REVIEW

File: teloxide_tests/src/mock_bot.rs [hunk 1]
```diff
@@ -346,6 +319,7 @@ impl MockBot {
         {
             left_tries -= 1;
             if left_tries == 0 {
+                cancel_token.cancel();
                 self.close_bot().await;
```
Comment (LasterAlex) [inline teloxide_tests/src/mock_bot.rs:323]: Why no `server.await.unwrap();` here?

Comment (LasterAlex) [reply teloxide_tests/src/mock_bot.rs:323]: Also, i am now wondering, is it possible, using `server`, to check if it has started...

**Review by LasterAlex — COMMENTED** [review]
Looks good! And the tests are x2 as fast now

File: teloxide_tests/src/mock_bot.rs (hunk 1 above)
Comment (will-lynas) [reply teloxide_tests/src/mock_bot.rs:323]: honestly, I don't know why this one doesn't need it and the other one does. It seems to work for the moment. #19 should make this all a lot simpler by putting the server cleanup in a drop method.

**Comment by will-lynas** [comment]
These changes will be included in #19

**Pull request closed without merging**